openpype/tools/loader/widgets.py

```diff
@@ -584,14 +584,15 @@ def on_context_menu(self, point):
             for repre_doc in repre_docs:
                 repre_ids.append(repre_doc["_id"])
 
+                # keep only version ids without representation with that name
                 version_id = repre_doc["parent"]
-                if version_id not in version_ids:
-                    version_ids.remove(version_id)
+                version_ids.remove(version_id)
 
-            for version_id in version_ids:
+            if version_ids:
+                # report versions that didn't have valid representation
                 joined_subset_names = ", ".join([
-                    '"{}"'.format(subset)
-                    for subset in subset_names_by_version_id[version_id]
+                    '"{}"'.format(subset_name_by_version_id[version_id])
+                    for version_id in version_ids
```

Comment:
Now this prints just reports just a single `Subsets x, y, z don't have representation 'name'` whereas previously it echoed it per subset name.